### ISSUE TITLE
ejabberd 18.03 on Raspberry Pi 2, ejabberdctl fails due to missing libtinfo.so.6

### ISSUE THREAD

**Issue by dognodding** [processone/ejabberd#2381]
> What version of ejabberd are you using?

v18.03

> What operating system (version) are you using?

Raspbian Stretch Lite Version: March 2018, Release date: 2018-03-13:
https://www.raspberrypi.org/downloads/raspbian/
Ubuntu 16.04 LTS 'classic' image for the Pi 2:
https://wiki.ubuntu.com/ARM/RaspberryPi

> How did you install ejabberd (source, package, distribution)?

Linux ARMv7 binary installer from ProcessOne from:
https://www.process-one.net/en/ejabberd/downloads/

> What did not work as expected? What was the unexpected behavior?

After installation with the binary installer, when I try to use ejabberdctl to add a user or start/stop ejabberd with fails with the following error:

`/opt/ejabberd-18.03/bin/beam.smp: error while loading shared libraries: libtinfo.so.6: cannot open shared object file: No such file or directory`

When I do `sudo find / -name 'libtinfo.so*' | more` I get the following ouput:

```
/usr/lib/arm-linux-gnueabihf/libtinfo.so
/lib/arm-linux-gnueabihf/libtinfo.so.5.9
/lib/arm-linux-gnueabihf/libtinfo.so.5
```

That is, libtinfo.so.6 seems not to be on the system but required by ejabberd (or at least by ejabberdctl).

I have tried this both with the latest Raspbian Stretch Lite image and also with the Ubuntu 16.04 LTS 'classic' image for the Pi.

Finally, I also installed ejabberd v18.01, but the same happens.

> Additional information

I have successfully installed ejabberd v18.03 on an Intel system, using the 64bit x86 binary installer. On this system, ejabberd works without any problems. When I do the same `sudo find / -name 'libtinfo.so*' | more` on my Intel system, the output is:

```
/opt/ejabberd-18.03/lib/linux-x86_64/libtinfo.so.5
/lib/x86_64-linux-gnu/libtinfo.so.5.9
/lib/x86_64-linux-gnu/libtinfo.so.5
```

That is, on the working x86_64 system, libtinfo.so.6 doesn't seem to be needed.

> Are there error messages in the log?

No. As ejabberd does not start, there is no log file.

> What was the expected result?

ejabberd should start without throwing an error. Is there something I can do do convince ejabberdctl to work with libtinfo.so.5 on the Raspberry Pi, as it seems to be able to do on an Intel machine?

**Comment by dognodding** [comment]
Additional information:
On the Pi (ARMv7 installer), the contents of `/opt/ejabberd-18.03/lib/linux-armhf/` is:
libcrypto.so.1.0.0  libexpat.so.1.6.7  libcursesw.so.6  libssl.so.1.0.0  libyaml-0.so.2

On the Intel machine (x86_64 installer), the contents of `/opt/ejabberd-18.03/lib/linux-x86_64/` is:
libcrypto.so.1.0.0  libexpat.so.1.6.0  libcursesw.so.5  libssl.so.1.0.0  libtinfo.so.5  libyaml-0.so.2

That is, the binary installer for x86_64 includes the necessary file, even though it is present on the system. The ARMv7 installer does not include (a higher version of) the necessary file, where it would be needed as the system does not have it.

**Comment by dognodding** [comment]
Apologies, I should have mentioned that the OS used on the Intel machine I mentioned above (where ejabberd works perfectly) is Ubuntu 16.04.4.

**Comment by cromain** [comment]
ARM builds are done on systems with more bleeding edge version of libraries. I'll include libtinfo to the pacakge, so it works with raspbian as well.

**Comment by cromain** [comment]
libtinfo.so.6 added in arm packaging

**Comment by dognodding** [comment]
Thank you! I'll try it.

**Comment by cromain** [comment]
this will be included in 18.03 installers (i won't re-generate 18.03)

**Comment by dognodding** [comment]
Just to let you know, I installed v18.04 on the Raspberry Pi (with raspbian OS) and everything is working. Thank you for fixing this.
